syntax = "proto3";

package dip_q;

option java_package = "cruz.agents";

/**
It's important that the communication between modules remains as simple as possible.
Therefore, in this message, we are ONLY sending and receiving arrays of integers.
**/

message Message {

  enum Type {
    INVALID = 0;
    GET_DEAL_REQUEST = 1;
    GET_ACCEPTANCE_REQUEST = 2;
  }

  // Using "required" is often bad practice. Generally, the validation
  // should be application-specific in order to make maintenance easier.
  // In proto3 syntax, "optional" is the default field

  Type type = 1;
  ObservationData observation = 2;
  DealData deal = 3;
  AcceptanceData acceptance = 4;
}

message ProvinceData {
  int32 id = 1;
  int32 owner = 2;
  int32 sc = 3;
}

message ObservationData {
  repeated ProvinceData provinces = 1;
  float previousActionReward = 2;
  bool done = 3;
  string info = 4;
}

message AcceptanceData {

}

message DealData {
  int32 powerToPropose = 1;
  int32 startProvince = 2;
  int32 destinationProvince = 3;
}